bandwidthd-php: PHP files to graph bandwidthd data in a postgresql database

padre-lacroix opened the pull request:
Maintainer: Jean-Michel Lacroix / @padre-lacroix
Compile tested: nothing really to compile: just a set of PHP files.  Compile tested on LEDE r857 for arm_cortex-a9_vfpv3 to generate a package that will also install the required packages to generate the graphs
Run tested: LEDE r857 on a WRT1900AC V1 (arm_cortex-a9_vfpv3).  Used and graphed the data generated by bandwidthd-pgsql on another OpenWRT router (TP-WDR3600 running 15.05) and stored in a postgresql database on a CentOS 6 server.  Graph are generated correctly.  The PHP files in bandwidthd-php were also used for the last few months by the CentOS server to graph the bandwidthd data.

Description:  bandwidthd-php is a set of PHP files that allows to graph bandwidthd
data stored in a postgresql database.  This bandwidthd data can be
generated using bandwidthd-pgsql in OpenWRT or LEDE or with
bandwidthd on different OS (Windows, Linux for example).

Signed-off-by: Jean-Michel Lacroix <lacroix@lepine-lacroix.info>
bandwidthd-php is a set of PHP files that allows to graph bandwidthd
data stored in a postgresql database.  This bandwidthd data can be
generated using bandwidthd-pgsql in OpenWRT or LEDE or with
bandwidthd on different OS (Windows, Linux for example).

champtar commented on utils/bandwidthd-php/files/config.conf:
```diff
@@ -0,0 +1,7 @@
+define("DFLT_HEIGHT", 256);
+define("DFLT_INTERVAL", INT_DAILY);
+
+$db_connect_string = "host=192.168.1.1 user=postgres dbname=bandwidthd"
```
127.0.0.1 make more sense here?
also this config is lost on upgrade
https://wiki.openwrt.org/doc/devel/packages#packageconffiles_optional

padre-lacroix commented:
You raise good points here: the config file there is basically the config file straight from Bandwidthd as downloaded.  I just modified the various files to change the short PHP tags to regular PHP tags, which does not required the user to change its php.ini file when install php5.
As for the config file, the default from bandwidthd as downloaded does not have the host part as the default is to run it on the same host as the postgresql server.  I added the host item to allow easily to configure on which host is the postgresql database.  Postgresql is quite huge and some people may use their OpenWRT/LEDE router to collect bandwidthd data with the package bandwidthd-pgsql, put store it to postgresql on another server, which prevent loosing the data when the router is rebooted: a router does not have really the space for such data, unless one install an external drive.  Then they can use the rate OpenWRT/LEDE router to display the graphs by getting the data on the external postgresql database.
There are many configuration possible with bandwidthd-pgsql and bandwidthd-php: each piece (collecting the data, storing in a postgresql database and creating the graphs) can be on a different system on on the same system.
I guess that I can create a config file in /etc/config and an init file in /etc/init.d to configure the config.conf file that is provided in the bandwidthd-php package.  This way, the configuration would not be lost when upgrading.
I will work on this in the next few days and perform another commit.  But, I do not promise a procd init file as I still am not sure on how to write one: I am trying to find good documentation on this.
Thanks for the suggestion!

padre-lacroix commented:
OK: slowly I am getting there!
I have now a configuration file (bandwidthd_php in /etc/config) and an init file (bandwidthd_php in /etc/init.d) that creates the config file above, but at a different location: /var/bandwidthd-php.conf and I replace the file /www/phphtdocs.config.conf by a symlink to /var/bandwidthd-php.conf.
Running the init script generates the file /var/bandwidthd-php.conf.
This is working fine, but I need to perform some testing to add to my init file some checks in case some options are missing in /etc/config/bandwidthd_php: I need to provide default values: this is the work that I have left, with the exception of modifying it to have a procd init scripts (am am reading on this).
So, this message is just to say that I am working on it and also to check whether what I did make sense: generating a config file, with the init script, in /var.
I hope to have everything ready to perform a commit (that will be squashed with the original one) in about 10 days (faster if I have the time).

padre-lacroix commented:
OK: I added a config file and an init file (using procd) as described above.  I squashed the two commits into one and updated the push.
Let me know if everything is to your satisfaction.

thess commented:
LGTM - thnx
